lib/_http_incoming.js

```diff
@@ -63,6 +61,16 @@ util.inherits(IncomingMessage, Stream.Readable);
 
 exports.IncomingMessage = IncomingMessage;
 
+Object.defineProperty(IncomingMessage.prototype, 'client', {
+  configurable: true,
+  enumerable: true,
+  get: util.deprecate(function() {
+    return this._client;
+  }, 'client is deprecated, use socket or connection instead'),
+  set: util.deprecate(function(val) {
+    this._client = val;
+  }, 'client is deprecated, use socket or connection instead')
+});
```

Comment:
Of course, this will break if anyone is trying to do `delete req.client` for whatever reason..